File: src/ProjectR/content/scripts/spells/HellfireFlare.lua
function GetTargetType()
  return Enemies
end

function GetName()
  return "Hellfire Flare"
end

function GetDescription()
  return "Shoot a massive ball of hellfire at your enemies.\nHuge delay."
end

function IsSupportSpell()
    return false
end

function GetMPCost()
    return 1.68
end

function GetDelay()
  return 0
end

function GetSpellType()
  return Pure
end

function GetMasteries()
  return { FIR }
end

function SpellEffect()
  local damage = 3.5 * aMD * (aFIR/100)
  d:TakeDamage(damage)
end

File: src/ProjectR.Scripting/content/scripts/spells/HellFireFlare.cs
using System.Collections.Generic;
using ProjectR.Interfaces;
using ProjectR.Interfaces.Model;
using ProjectR.Interfaces.Model.Stats;

namespace ProjectR.Scripting.Spells
{
    public class HellFireFlare : SpellScriptBase
    {
        public override string Name { get { return "Hellfire Flare"; } }
        public override string Description { get { return "Shoot a massive ball of hellfire at your enemies.\nHuge delay."; } }

        public override TargetType TargetType { get { return TargetType.Enemies; } }
        public override IList<EleMastery> Masteries { get { return new EleMastery[] { EleMastery.FIR, }; } }
        public override SpellType SpellType { get { return SpellType.Pure; } }
        public override bool IsSupportSpell { get { return false; } }

        public override double MPCost { get { return 168; } }
        public override double Delay { get { return 0; } }

        public override void SpellEffect(ICharacter caster, ICharacter target)
        {
            var damage = 3.5 * cMD * (cFIR / 100);
            DealDamage(damage);
        }
    }
}